### PR TITLE
feat: create a job for scheduling reports and the controller endpoint for it

### PR DESCRIPTION
**Pull request by MoralesWilll**
What does this do

- Setup sidekiq to support background jobs
- Create a job with sidekiq to manage the creation of reports at intervals
- Add the endpoint to use the job

### CODE REVIEW

File: app/sidekiq/report_scheduler_job.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    end
+  end
+
+  # Helper method to reschedule the job at a fixed interval
+  def schedule_next_job(interval_in_seconds, event_id, frequency)
+    # Calculate the next run time by adding the interval (in seconds) to the current time
+    next_run_time = Time.now + interval_in_seconds
+
+    # Schedule the next job at the exact time (this keeps it consistent)
+    self.class.perform_at(next_run_time, event_id, frequency)
```
Comment (daguttt) [inline app/sidekiq/report_scheduler_job.rb:47]: @MoralesWilll Using `.perform_at` creates a **recurrent** job?

File: config/application.rb [hunk 2]
```diff
@@ -34,11 +34,11 @@ class Application < Rails::Application
...
     #
     # config.time_zone = "Central Time (US & Canada)"
-    # config.eager_load_paths << Rails.root.join("extras")
-
+    # config.eager_load_paths << Rails.root.join("extras")"
     # Only loads a smaller set of middleware suitable for API only apps.
     # Middleware like session, flash, cookies can be added back manually.
     # Skip views, helpers and assets when generating a new resource.
     config.api_only = true
+    config.middleware.use ActionDispatch::Session::CookieStore, key: "_your_session_key", secret: "509e24d6b81dc8e517bac3da452f45ab8f6a55fc09c655f561023da6b485b2119a12ce788ffc841e7437981180ce7ee9550c6529ff08e3a31d85ef070bf978c8"
```
Comment (daguttt) [inline config/application.rb]: @MoralesWilll Create an environment variable for session key

Comment (MoralesWilll) [reply config/application.rb]: Will do

File: app/sidekiq/report_scheduler_job.rb (hunk 1 above)
Comment (MoralesWilll) [reply app/sidekiq/report_scheduler_job.rb:47]: Not by itself, it just schedules itself in the future using the parameters, but when I call that it becomes recurrent, yes

**Review by daguttt — APPROVED** [review]
Nice!

**Pull request merged**